src/app/AttributeValueDecoder.h

```diff
@@ -52,6 +52,15 @@ class AttributeValueDecoder
         return CHIP_NO_ERROR;
     }
 
+    /**
+     * Get the reader from the decoder for raw data. Calling this is considered to be tried decoding.
```

Comment:
We pretty explicitly did not want to hand out the reader from the decoder, because it's too easy to misuse.

Why is this actually needed?  The "multi-threaded case" comment from the PR description sounds like "misuse"...

Reply:
If so, what do you think about returning a ByteSpan? We intended to copy the data to a different thread, not intended to modify the data.

Reply:
The use case seems to be "I want the data, but don't actually know what it contains and want an opaque blob" ... is this true?

Would you be able to implement that by creating a class for which DataModel::Decode exists and creates whatever is necessary? Decode does receive the TLV object.

Reply:
Thank you. That is the use case that I need. I will try it.

Reply:
Ah, yes, if you're going to snapshot the data then it should be fine.  This sounds like PreEncodedValue but in the other direction...

(That said, the TLV reader on the other thread which then tries to read the data will need to be handled very carefully.... and for some attributes you need a bunch of state other than the data to decode properly.)